lib/endpoints/class-wp-rest-controller.php

```diff
@@ -127,10 +127,12 @@ public function delete_item( $request ) {
 	/**
 	 * Prepare the item for create or update operation.
 	 *
-	 * @param WP_REST_Request $request Request object.
-	 * @return WP_Error|object $prepared_item
+	 * @param WP_REST_Request $request  Request object.
+	 * @param bool            $creating Whether the item is being created.
```

Comment:
> Can we follow this pre-existing pattern?

Good point, I'll make those changes.

> Also, do we need to necessarily pass a variable to the method, or is there a way of determining whether it's a create or update already from within the method? I think it might be safe to key off whether or not the ID is present on the prepared database object.

This looks like it will work for all of the subclasses of `WP_REST_Controller` except for `WP_REST_Comments_Controller` and `WP_REST_Terms_Controller`. The `prepare_item_for_database()` method for these classes does not handle the ID field. For comments, it's possible this was done to facilitate an easy way to determine if only the comment `status` is changing in `update_item()`. Perhaps the logic could be updated to allow for an ID being set as well as the comment status?

As for the terms, maybe there's room for an adjustment there as well? I'm seeing this line twice:

```php
$term = get_term( (int) $request['id'], $this->taxonomy );
```